paasta_tools/autoscaling_lib.py

```diff
@@ -775,19 +778,18 @@ class ClusterAutoscalingError(Exception):
     pass
 
 
-def get_instance_id_from_ip(ip):
+def get_instance_id_from_ip(ip, instance_descriptions):
```

Comment:
This whole function should be far less opinionated about what it's doing.

See the following:

```
def find_instance_for_ip(instance_descriptions, ip):
     return [instance for instance in instance_descriptions if instance['PrivateIpAddress']
```
Then the caller can decide what to do according to the return value:

```
instances = find_instance_for_ip(descriptions, ip)
if not instances:
   log.warning("Couldn't find instance ID for ip {0}".format(ip))
   //I can do something here now that there are none
if len(instances) > 1:
   log.warning("wtf how did this happen")
else:
    //only care about the id for now, but the whole instance object is here should
    //I need something else from it later
    instance_id = instances[0]['InstanceId']
```

See how the function now cares far less about what the result of the match 'means' (no matches, >1 match, etc), and what information in the return value is useful (only the ``instanceid`` before) and leaves it to the caller to care instead? It makes it more reusable.